frontend/public/components/factory/table.tsx

```diff
@@ -125,8 +135,11 @@ const stateToProps = ({UI}, {data = [], defaultSortField = 'metadata.name', defa
   if (loaded) {
     let sortBy: string | Function = 'metadata.name';
     if (currentSortField) {
-      // Sort resources by one of their fields as a string
-      sortBy = resource => sorts.string(_.get(resource, currentSortField, ''));
+      // Sort resources by one of their fields as a string or number
+      sortBy = resource => {
+        const sortFldValue = _.get(resource, currentSortField, '');
+        return _.isFinite(sortFldValue) ? sorts.number(sortFldValue) : sorts.string(sortFldValue);
```

Comment:
I don't think we want to treat strings as numbers. It would be better to have a custom sort func that handles the conversion when we know for sure it's always a numeric string. This could break sorts for data where some values are strings that happen to be numbers and some aren't.

Reply:
yeah, I realize this, but didn't think a column of mixed numbers and strings is common.  I tried using the singe `sortFunc' for every column which has numberic data, but that returns the entire row in the funt, and I can't determine which column to actually sort on.  The alternative would be to create a unique `sortFunc` for every possible chargeback report numeric column.  CC @priley86

Reply:
i've used natural sort in the past for cases like this, but i don't know if that's intended here (or if you can prevent mixing those types). That being said, if it's unknown, a natural sort might be useful as a generic sort method at some point...

https://github.com/ManageIQ/manageiq-v2v/blob/8598e906d308777252bd7b5604f6412d71327cdf/app/javascript/react/screens/App/common/ListViewToolbar/sortFilter.js#L25

https://en.wikipedia.org/wiki/Natural_sort_order

Reply:
Hi, modified it so that if sortField AND sortFunc are defined, then sortField is passed to sortFunc.  This way I can create a generic sortFunc which will process any sortField also defined -instead of having to define named sortFunctions.